lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -63,6 +59,94 @@ struct ConnectInfo {
 using InnerRefMap =
     DenseMap<hw::InnerRefAttr, std::pair<hw::InnerSymAttr, StringAttr>>;
 
+//===----------------------------------------------------------------------===//
+// Naming Helpers
+//===----------------------------------------------------------------------===//
+
+/// For a layer `@A::@B::@C` in module Module,
+/// the generated module is called `Module_A_B_C`.
+static SmallString<32> moduleNameForLayer(StringRef moduleName,
+                                          SymbolRefAttr layerName) {
```

Comment:
General nit: passing a `SmallString` reference that is mutated may be cleaner to do here.